xenomint/pool.go

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2018 The CovenantSQL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package xenomint
+
+import (
+	"sync"
+
+	"github.com/CovenantSQL/CovenantSQL/types"
+)
+
+type QueryTracker struct {
+	sync.RWMutex
+	Req  *types.Request
+	Resp *types.Response
+}
+
+func (q *QueryTracker) UpdateResp(resp *types.Response) {
+	q.Lock()
+	defer q.Unlock()
+	q.Resp = resp
+}
+
+func (q *QueryTracker) Ready() bool {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method QueryTracker.Ready should have comment or be unexported